CRM/Case/BAO/Case.php

```diff
@@ -1041,7 +1041,10 @@ public static function getCaseActivity($caseID, &$params, $contactID, $context =
 
     // define statuses which are handled like Completed status (others are assumed to be handled like Scheduled status)
     $compStatusValues = array();
-    $compStatusNames = array('Completed', 'Left Message', 'Cancelled', 'Unreachable', 'Not Required');
+    $compStatusNames = array_merge(
+      array('Completed', 'Left Message', 'Cancelled', 'Unreachable', 'Not Required'),
+      CRM_Activity_BAO_Activity::getStatusesByType(CRM_Activity_BAO_Activity::COMPLETED)
+    );
```

Comment:
@jitendrapurohit why are you merging these 2 arrays? I think just the statusesByType should provide the needed information, correct?

Reply:
doesn't seem so. 'Left Message', 'Unreachable' and 'Not Required' belongs to activity statuses of type `INCOMPLETE` and 'Cancelled' is of type `CANCELLED`. 

But the current flow doesn't want to display them in red color if they have an overdue date(see comment in L1042). statusesByType will only return the completed activity status.

We can use the following API if we intend to use only one variable -

    $compStatusValues = civicrm_api3('OptionValue', 'get', array(
      'sequential' => 1,
      'return' => array("value"),
      'option_group_id' => "activity_status",
      'name' => array('IN' => array("Left Message", "Cancelled", "Unreachable", "Not Required")),
      'filter' => CRM_Activity_BAO_Activity::COMPLETED,
      'options' => array('or' => array(array("name", "filter"))),
    ));
    $compStatusValues = CRM_Utils_Array::collect('value', $compStatusValues['values']);

But seems it'll not be a good change w.r.t readability than the current one?